ttg/ttg/terminal.h

```diff
@@ -11,8 +11,66 @@
 #include "ttg/util/meta.h"
 #include "ttg/util/trace.h"
 #include "ttg/world.h"
+#include "boost/callable_traits.hpp"
 
 namespace ttg {
+  namespace detail {
+
+    /* Wraps any key,value data structure.
+     * Elements of the data structure can be accessed using get method, which calls the at method of the Container.
+     * keyT - taskID
+     * valueT - Value type of the Container
+    */
+    template<typename keyT, typename valueT>
+    struct ContainerWrapper {
+      std::function<valueT (keyT const& key)> get = nullptr;
+      std::function<size_t (keyT const& key)> owner = nullptr;
+
+      ContainerWrapper() = default;
+
+      template<typename T, typename mapperT, typename keymapT, std::enable_if_t<!std::is_same<std::decay_t<T>,
+                                                          ContainerWrapper>{}, bool> = true>
+        //Store a pointer to the user's container in std::any, no copies
+        ContainerWrapper(T &t, mapperT &&mapper,
+                         keymapT &&keymap) : get([&t,mapper = std::forward<mapperT>(mapper)](keyT const &key) {
```

Comment:
I'm a little worried that if we don't forward `t` we risk losing the lambda. And why do we only forward the mapper and not the keymap? Maybe it should be:

```C
ContainerWrapper(T &&t, mapperT &&mapper,
                         keymapT &&keymap) : get([t = std::forward<T>(t),
                                                  mapper = std::forward<mapperT>(mapper)
                                                  keymap = std::forward<keymapT>(keymap)](keyT const &key) {
```